sources/packages/backend/libs/integrations/src/services/disbursement-schedule/e-cert-generation.service.ts

```diff
@@ -0,0 +1,506 @@
+        "sinValidation.sin",
+        "student.id",
+        "student.birthDate",
+        "student.gender",
+        "student.contactInfo",
+        "institutionLocation.id",
+        "institutionLocation.institutionCode",
+        "disbursementValue.id",
+        "disbursementValue.valueType",
+        "disbursementValue.valueCode",
+        "disbursementValue.valueAmount",
+        "disbursementValue.disbursedAmountSubtracted",
+        "studentAssessment.id",
+      ])
+      .addSelect(
+        `CASE
+            WHEN EXISTS(${this.studentRestrictionService
+              .getExistsBlockRestrictionQuery(
+                false,
+                false,
+                "restrictionActionType",
+              )
+              .getSql()}) THEN true
+            ELSE false
+        END`,
+        "stopFullTimeBCFunding",
+      )
+      .innerJoin("disbursement.studentAssessment", "studentAssessment")
+      .innerJoin("studentAssessment.application", "application")
+      .innerJoin("application.currentAssessment", "currentAssessment") // * This is to fetch the current assessment of the application, even though we have multiple reassessments
+      .innerJoin("currentAssessment.offering", "offering")
+      .innerJoin("offering.institutionLocation", "institutionLocation")
+      .innerJoin("offering.educationProgram", "educationProgram")
+      .innerJoin("application.student", "student") // ! The student alias here is also used in sub query 'getExistsBlockRestrictionQuery'.
+      .innerJoin("student.user", "user")
+      .innerJoin("student.sinValidation", "sinValidation")
+      .innerJoin("application.msfaaNumber", "msfaaNumber")
+      .innerJoin("disbursement.disbursementValues", "disbursementValue")
+      .where("disbursement.dateSent is null")
+      .andWhere("disbursement.disbursementDate <= :disbursementMinDate", {
+        disbursementMinDate,
+      })
+      .andWhere("application.applicationStatus = :applicationStatus", {
+        applicationStatus: ApplicationStatus.completed,
+      })
+      .andWhere("msfaaNumber.dateSigned IS NOT NULL")
+      .andWhere("msfaaNumber.cancelledDate IS NULL")
+      .andWhere("sinValidation.isValidSIN = true")
+      .andWhere("offering.offeringIntensity = :offeringIntensity", {
+        offeringIntensity,
+      })
+      .andWhere(
+        `NOT EXISTS(${this.studentRestrictionService
+          .getExistsBlockRestrictionQuery()
+          .getSql()})`,
+      )
+      .andWhere(
+        "disbursement.disbursementScheduleStatus = :disbursementScheduleStatus",
+        {
+          disbursementScheduleStatus: DisbursementScheduleStatus.Pending,
+        },
+      )
+      .andWhere("disbursement.coeStatus = :coeStatus", {
+        coeStatus: COEStatus.completed,
+      })
+      .setParameter("restrictionActionType", stopFullTimeBCFunding)
+      // 'restrictionActions' parameter used inside sub-query.
+      .setParameter("restrictionActions", possibleRestrictionActions)
```

Comment:
Is this parameter being used in query?

Reply:
It is used in the subquery.

![image](https://user-images.githubusercontent.com/61259237/211693389-5090d020-b7a7-4193-ada3-c2b9f51c337f.png)

![image](https://user-images.githubusercontent.com/61259237/211693450-cd699113-2a43-418a-b02b-1062bfdafa2a.png)

Maybe in the future there will an opportunity to refactor this entire query 😉

Reply:
Ah I see.. I was tricky to understand.